docs/tutorials/docker.md

```diff
@@ -21,67 +21,108 @@ title: "Docker"
+
+The default values are fine for the Quickstart. Production systems will want to modify the defaults.
 
-Special environment variables:
+Basic configuration:
+
+* `DRUID_MAXDIRECTMEMORYSIZE` -- set Java max direct memory size. Default is 6 GB.
+* `DRUID_XMX` -- set Java `Xmx`, the maximum heap size. Default is 1 GB.
+
+Production configuration:
 
-* `JAVA_OPTS` -- set java options
-* `DRUID_LOG4J` -- set the entire `log4j.xml` verbatim
-* `DRUID_LOG_LEVEL` -- override the default log level in default log4j
-* `DRUID_XMX` -- set Java `Xmx`
-* `DRUID_XMS` -- set Java `Xms`
-* `DRUID_MAXNEWSIZE` -- set Java max new size
-* `DRUID_NEWSIZE` -- set Java new size
-* `DRUID_MAXDIRECTMEMORYSIZE` -- set Java max direct memory size
 * `DRUID_CONFIG_COMMON` -- full path to a file for druid 'common' properties
 * `DRUID_CONFIG_${service}` -- full path to a file for druid 'service' properties
+* `JAVA_OPTS` -- set Java options
+
+Logging configuration:
+
+* `DRUID_LOG4J` -- set the entire [`log4j.xml` configuration file](https://logging.apache.org/log4j/2.x/manual/configuration.html#XML)  verbatim. ([Example](https://github.com/apache/druid/blob/{{DRUIDVERSION}}/distribution/docker/environment#L52))
+* `DRUID_LOG_LEVEL` -- override the default [Log4j log level](https://en.wikipedia.org/wiki/Log4j#Log4j_log_levels)
+
+Advanced memory configuration:
+
+* `DRUID_XMS` -- set Java [`Xms`](https://docs.oracle.com/cd/E19900-01/819-4742/abeik/index.html), the initial heap size. Default is 1 GB.
+* `DRUID_MAXNEWSIZE` -- set [Java max new size](https://docs.oracle.com/cd/E19900-01/819-4742/abeik/index.html)
+* `DRUID_NEWSIZE` -- set [Java new size](https://docs.oracle.com/cd/E19900-01/819-4742/abeik/index.html)
 
-In addition to the special environment variables, the script which launches Druid in the container will also attempt to use any environment variable starting with the `druid_` prefix as a command-line configuration. For example, an environment variable
- 
-```druid_metadata_storage_type=postgresql```
- 
-would be translated into 
+In addition to the special environment variables, the script which launches Druid in the container will use any environment variable starting with the `druid_` prefix as command-line configuration. For example, an environment variable
 
-```-Ddruid.metadata.storage.type=postgresql```
- 
-for the Druid process in the container.
+`druid_metadata_storage_type=postgresql`
+
+is translated into the following option in the Java launch command for the Druid process in the container:
+
+`-Ddruid.metadata.storage.type=postgresql`
+
+Note that Druid uses port 8888 for the console. This port is also used by Jupyter and other tools. To avoid conflicts, you can change the port in the [ports](https://github.com/apache/druid/blob/0.21.1/distribution/docker/docker-compose.yml#L125) section of the `docker-compose.yml` file. Example to expose the Console on port 9999 of the host:
```

Comment:
```suggestion
Note that Druid uses port 8888 for the console. This port is also used by Jupyter and other tools. To avoid conflicts, you can change the port in the [`ports`](https://github.com/apache/druid/blob/0.21.1/distribution/docker/docker-compose.yml#L125) section of the `docker-compose.yml` file. For example, to expose the console on port 9999 of the host:
```